.github/workflows/images.yaml

```diff
@@ -0,0 +1,51 @@
+# NUC image creation and push to DockerHub
+name: NUC Image Push
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push events for the main branch
+  push:
+    branches: [main]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Build and push all images
+  build_and_push:
+    name: "Build and Push Docker Images"
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
```

Comment:
Reading in the `runs-on` parameter from repository environment/secrets would allow us to switch from hosted to self-hosted runners and vice versa easily by just changing the value in our repo settings. Might be handy.

Reply:
You really should try to avoid having a situation where an environment outside of the code can influence the building success of the code.

It's better to have this configuration in the code here so that if changes are needed in order to make the code build, they go through the code review process. The reason that things like the docker password aren't in here is you really don't want to commit the passwords to the code.

Reply:
Yeah ok makes sense, I was just thinking of a situation where we would suddenly want to switch where all our runners are running (Uni loses net/power etc), but I guess merging a PR into main, and merging main into whatever other branches is the better solution.